client/src/tests/App.test.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { render, fireEvent } from 'react-testing-library';
+import 'jest-dom/extend-expect';
+
+import App from '../components/App';
+
+
+
+describe('<App />', () => {
+
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<App />, div);
+    ReactDOM.unmountComponentAtNode(div);
+  });
+  
+  it('renders successfully', () => {
+    render(<App />);
+  })
+
+  // it('renders Hits!', () => {
+  //   const { queryByText, debug } = render(<App />);
+  //   debug();
+  //   expect(queryByText(/Hits: /i)).not.toBeNull();
+  // })
+
+  it('Foul counter works', () => { 
+    const { getByText } = render(<App />);
+    const button = getByText(/foul/i);
+
+    fireEvent.click(button);
+
+    getByText(/\d/i);
```

Comment:
this will return a passing case if it finds _any_ digit character on the page, so it will always pass. check to make sure you can successfully capture the initial display before the button is clicked by checking for /0 0/ (since initial is 0 balls and strikes), then after firing this event once, check for /0 1/ to test this specific behavior the button should be causing

Reply:
since you display the total number of foul balls, your could also run a matcher check for /foul balls: 0/i initially, then after the event fires, /foul balls: 1/i